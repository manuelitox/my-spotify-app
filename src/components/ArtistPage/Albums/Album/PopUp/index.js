import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import Track from 'components/generic/Track/'
import PopUp from 'components/generic/PopUp/container'
import PlaceholderAlbum from 'svgs/placeholderAlbum.svg'
import StylesHeaderAlbum from './styles-header'
import StylesWrapperTracks from './styles-wrapper-tracks'

const PopUpAlbum = ({ isOpen, togglePopUp, cover, title, releaseDate, numberTracks }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp }>
    <Fragment>
      <StylesHeaderAlbum>
        <img 
          src={ cover ? cover : PlaceholderAlbum } 
          alt={ title }
          width="300"
          height="300" />
        <h3>{ title }</h3>
        <p>Total tracks: { numberTracks }</p>
        <p>Release date: { releaseDate }</p>
      </StylesHeaderAlbum>
      <StylesWrapperTracks>
        <Track 
          name="Empire State of Mind" 
          duration="3 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="No one" 
          duration="4 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />
        <Track 
          name="Girl of Fire" 
          duration="3:23 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />    
        <Track 
          name="Fallin'" 
          duration="5:42 min"
          previewUrl="https://p.scdn.co/mp3-preview/d7624ec5f93b6d92c1836a95c40ecce463584f6e?cid=774b29d4f13844c495f206cafdad9c86"
          isPopUpVersion
          songIsAvailable />                
      </StylesWrapperTracks>
    </Fragment>
  </PopUp>      
)

PopUpAlbum.propTypes = {
  isOpen: PropTypes.bool,
  togglePopUp: PropTypes.func.isRequired,
  cover: PropTypes.string,
  title: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  numberTracks: PropTypes.number.isRequired
}

export default PopUpAlbum