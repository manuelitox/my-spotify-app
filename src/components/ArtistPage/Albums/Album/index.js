import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'

import PopUpAlbum from './PopUp/'
import StylesAlbum from './styles'
import PlaceholderAlbum from 'svgs/placeholderAlbum.svg'

class Album extends Component {
  state = { isOpen: false }

  togglePopUp = () => this.setState({ isOpen: !this.state.isOpen })

  render () {
    const { cover, title, releaseDate, numberTracks } = this.props
    return (
      <Fragment>
        <StylesAlbum onClick={ this.togglePopUp }>
          <img 
            src={ cover ? cover : PlaceholderAlbum } 
            alt={ title }
            width="60"
            height="60" />
          <div>
            <h3>{ title }</h3>
            <p>Total tracks: { numberTracks }</p>
            <p>Release date: { releaseDate }</p>
          </div>
        </StylesAlbum>      
        <PopUpAlbum 
          isOpen={ this.state.isOpen } 
          togglePopUp={ this.togglePopUp }
          title={ title }
          releaseDate={ releaseDate }
          numberTracks={ numberTracks }/>
      </Fragment>
    )
  }
}

Album.propTypes = {
  cover: PropTypes.string,
  title: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  numberTracks: PropTypes.number.isRequired
}

export default Album