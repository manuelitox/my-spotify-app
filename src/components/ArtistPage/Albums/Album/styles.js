import styled from 'styled-components'

const StylesAlbum = styled.div`
  cursor: pointer;
  display: grid;
  grid-template-columns: 60px 2fr;
  grid-gap: 20px;
  img {
    border-radius: 4px;
  }
  h3 {
    color: white;
    font-size: 1.6rem;
    font-weight: 500;
    margin: 0 0 10px;
  }
  p {
    color: ${ props => props.theme.blueDarkText };
    font-size: 1.4rem;
    margin: 2px 0;
  }
` 

export default StylesAlbum