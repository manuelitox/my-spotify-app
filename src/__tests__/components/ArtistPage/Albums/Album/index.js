import React from 'react'
import { shallow } from 'enzyme'

import Album from 'components/ArtistPage/Albums/Album/'

describe('Component | Album', () => {
  let component 
  beforeEach( () => {
    component = shallow(
      <Album title="album title" releaseDate="2016-04-07" numberTracks={Â 10 } />
    )
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
  })
})