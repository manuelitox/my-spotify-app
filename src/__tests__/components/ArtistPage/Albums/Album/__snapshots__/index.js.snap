// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | Album should render a component 1`] = `
<Fragment>
  <styled.div
    onClick={[Function]}
  >
    <img
      alt="album title"
      height="60"
      src="placeholderAlbum.svg"
      width="60"
    />
    <div>
      <h3>
        album title
      </h3>
      <p>
        Total tracks: 
        10
      </p>
      <p>
        Release date: 
        2016-04-07
      </p>
    </div>
  </styled.div>
  <PopUpAlbum
    isOpen={false}
    numberTracks={10}
    releaseDate="2016-04-07"
    title="album title"
    togglePopUp={[Function]}
  />
</Fragment>
`;
